src/app/leaderboard/LeaderboardTable.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+import {
+  ColumnDef,
+  SortingState,
+  createColumnHelper,
+  getCoreRowModel,
+  getSortedRowModel,
+  flexRender,
+  useReactTable,
+} from "@tanstack/react-table";
+import Player from "../types/player";
+import { useReducer, useState } from "react";
+import React from "react";
+
+export const LeaderboardTable = ({ data }) => {
```

Comment:
Because we can't use the filesystem functions on the client we instead we pass the server's data to the client component via props